visual-dotnet/SauceLabs.Visual/VisualClient.cs

```diff
@@ -64,38 +67,53 @@ public VisualClient(WebDriver wd, Region region, string username, string accessK
         /// <param name="username">the Sauce Labs username</param>
         /// <param name="accessKey">the Sauce Labs access key</param>
         /// <param name="buildOptions">the options of the build creation</param>
-        public VisualClient(WebDriver wd, Region region, string username, string accessKey, CreateBuildOptions buildOptions)
+        public static async Task<VisualClient> Create(WebDriver wd, Region region, string username, string accessKey, CreateBuildOptions buildOptions)
         {
-            if (StringUtils.IsNullOrEmpty(username) || StringUtils.IsNullOrEmpty(accessKey))
-            {
-                throw new VisualClientException("Username or Access Key not set");
-            }
+            var client = new VisualClient(wd, Region.UsWest1, username, accessKey, buildOptions);
 
-            _api = new VisualApi<WebDriver>(wd, region, username, accessKey);
-            _sessionId = wd.SessionId.ToString();
-            _jobId = wd.Capabilities.HasCapability("jobUuid") ? wd.Capabilities.GetCapability("jobUuid").ToString() : _sessionId;
-            var response = _api.WebDriverSessionInfo(_jobId, _sessionId).Result;
+            var response = await client._api.WebDriverSessionInfo(client._jobId, client._sessionId);
             var metadata = response.EnsureValidResponse();
```

Comment:
idea: how about moving this code to private method and avoid setting internal object properties from the static method?